### PR TITLE
Update usePoolBalance to have one Subject per pool

### PR DESCRIPTION
**Pull request by lazarmitic**

### CODE REVIEW

File: packages/tempus-client_v3/src/components/PoolPositionModal/PoolPositionModalResolver.tsx [hunk 1]
```diff
@@ -1,23 +1,21 @@
...
   // TODO - Properly check if URL params have valid values - if not show an error page or redirect to root page
 
-  const poolChainAddressId = `${chain}-${poolAddress}`;
-
   return (
     // TODO - Replace dummy data with data from hooks
     <PoolPositionModal
       apr={0.1}
-      balance={poolBalances[poolChainAddressId]}
+      balance={poolBalanceData?.balance || new Decimal(0)}
```
Comment (iamvukasin) [inline packages/tempus-client_v3/src/components/PoolPositionModal/PoolPositionModalResolver.tsx:18]:
```suggestion
      balance={poolBalanceData?.balance || ZERO}
```

**Comment by danieledamiani** [comment]
in review

File: packages/tempus-client_v3/src/hooks/usePoolViewOptions.ts [hunk 2]
```diff
@@ -87,8 +87,15 @@ const filteredSortedPoolList$ = combineLatest([filteredPoolList$, stateSortType$
             return aprA.gt(aprB) ? factor : -1 * factor;
           }
           case 'balance': {
-            const poolBalancesA = poolBalances[`${poolA.chain}-${poolA.address}`] ?? ZERO;
-            const poolBalancesB = poolBalances[`${poolB.chain}-${poolB.address}`] ?? ZERO;
+            const poolABalanceData = poolBalances.find(
+              poolBalance => poolBalance.chain === poolA.chain && poolBalance.address === poolA.address,
+            );
+            const poolBBalanceData = poolBalances.find(
```
Comment (danieledamiani) [inline packages/tempus-client_v3/src/hooks/usePoolViewOptions.ts:93]: maybe something like `secondPoolBalance` would be more readable

File: packages/tempus-client_v3/src/hooks/usePoolViewOptions.ts [hunk 3]
```diff
@@ -87,8 +87,15 @@ const filteredSortedPoolList$ = combineLatest([filteredPoolList$, stateSortType$
             return aprA.gt(aprB) ? factor : -1 * factor;
           }
           case 'balance': {
-            const poolBalancesA = poolBalances[`${poolA.chain}-${poolA.address}`] ?? ZERO;
-            const poolBalancesB = poolBalances[`${poolB.chain}-${poolB.address}`] ?? ZERO;
+            const poolABalanceData = poolBalances.find(
```
Comment (danieledamiani) [inline packages/tempus-client_v3/src/hooks/usePoolViewOptions.ts:90]: a `find` inside a `sort` doesn't look very efficient.. I think moving `poolBalances` into a map `{ address: poolBalance}` would improve performances a bit

Comment (lazarmitic) [reply packages/tempus-client_v3/src/hooks/usePoolViewOptions.ts:90]: Converted array to map.

**Review by danieledamiani — APPROVED** [review]

**Pull request merged**